internal/hwmon/hwmon.go

```diff
@@ -162,27 +161,15 @@ func GetFans(chip gosensors.Chip) map[int]*fans.HwMonFan {
 		subfeatures := feature.GetSubFeatures()
 
 		if containsSubFeature(subfeatures, gosensors.SubFeatureTypeFanInput) {
-			pwmOutput := path.Join(chip.Path, fmt.Sprintf("pwm%d", currentOutputIndex+1))
-
-			if _, err := os.Stat(pwmOutput); err == nil {
-			} else if errors.Is(err, os.ErrNotExist) {
-				// path/to/whatever does *not* exist
-				pwmOutput = ""
-			} else {
-				pwmOutput = ""
-			}
-
-			currentOutputIndex++
-
-			if len(pwmOutput) <= 0 {
+			var channel int
+			_, err := fmt.Sscanf(feature.Name, "fan%d", &channel)
+			if err != nil {
 				continue
```

Comment:
Should we log an error, or at least a warning here?
Is it expected that the fan does not have a channel (or a number suffix I guess) associated to it?

Reply:
The Linux hwmon subsystem guarantees a numerical channel number.

https://www.kernel.org/doc/html/next/hwmon/sysfs-interface.html

> Numbering usually starts from 1, except for voltages which start from 0 (because most data sheets use this). A number is always used for elements that can be present more than once, even if there is a single element of the given type on the specific chip.

Reply:
Yes should work. If it doesn't I'll figure out a way later on.

Reply:
I have added a warning log.